### ISSUE TITLE
No Sound in Lego Lord of the Rings

### ISSUE THREAD

**Issue by dbordak** [ValveSoftware/steam-runtime#459]
#### Your system information

* Steam Runtime Version:  Soldier
* Distribution (e.g. Ubuntu 18.04): Arch Linux
* Link to your full system information (Help -> System Information) in a [Gist](https://gist.github.com/): https://gist.github.com/dbordak/d4fdc0ef5c579ce0acb851832f44a2e5
* Have you checked for system updates?: Yes
* What compatibility tool are you using?: Proton 5.13+
* If you are using Steam Linux Runtime for native Linux games: What versions are listed in `SteamLinuxRuntime/VERSIONS.txt`? N/A
* If you are using Steam Linux Runtime, or Proton 5.13 or newer: What versions are listed in `SteamLinuxRuntime_soldier/VERSIONS.txt`?
```
#Name	Version		Runtime	Runtime_Version	Comment
depot	0.20210818.24			# Overall version number
pressure-vessel	0.20210809.1			
scripts	v0.20210818.0-0-gc6be089			# Entry point scripts, etc.
soldier	0.20210817.0	soldier	0.20210817.0	# soldier_platform_0.20210817.0/
```

#### Please describe your issue in as much detail as possible:
Running the game on either the default branch or the `client_beta` branch of the Soldier runtime, when launching the game, I get no sound.
Running it with `previous_version`, it works as expected.
(the snippet i posted above is with it set to the default branch)
Reverting to proton 5.0, I still get no sound, so I'm not sure what to make of that.

Using `PROTON_LOG=1`, I can see this line in the logs:
```
INFO: SDL_INIT_AUDIO failed: Audio target 'pulse' not available
```

Upon finding that line, I tried `SDL_AUDIODRIVER=alsa`, which also appears to fix the issue.

I'm using pipewire-pulse (as well as pipewire-alsa), if that matters.

#### Steps for reproducing this issue:

1. launch the game.

**Comment by kisak-valve** [comment]
Hello @dbordak, please give #452 a read. `SDL_AUDIODRIVER=pulse` has never been a valid option for SDL 2, while it was an option for SDL 1.2. If you do need to have that environment variable set, then it would be `SDL_AUDIODRIVER=pulseaudio`.

SDL has fairly good default order of audio backends it uses. Is there a particular reason you're setting that environment variable at all?

**Comment by dbordak** [comment]
Thank you for the extremely prompt response. It took a while for me to figure out where that was being set (it was `~/.pam_environment`), but I have no idea _why_ it was set.

Sorry for the trouble, since it was my own environment after all!

**Comment by smcv** [comment]
Today's beta update for "Steam Linux Runtime - soldier" has a backport of [an SDL commit](https://github.com/libsdl-org/SDL/commit/3261f7f6ceaaa435a3aca0252b2f7126eeae1bb2) which makes `SDL_AUDIODRIVER=pulse` an alias for `SDL_AUDIODRIVER=pulseaudio`, avoiding the problem.

I would still recommend not setting the environment variable at all - that will give you SDL's default behaviour, trying several possible backends in a preference order, which is more likely to continue to do the right things in future than forcing it to use a particular backend.

To check which version you have, look at `SteamLinuxRuntime_soldier/VERSIONS.txt`. The version that fixes this issue is `depot 0.20210907.26` or later.

**Comment by smcv** [comment]
> Today's beta update for "Steam Linux Runtime - soldier" has a backport of [an SDL commit](https://github.com/libsdl-org/SDL/commit/3261f7f6ceaaa435a3aca0252b2f7126eeae1bb2) which makes `SDL_AUDIODRIVER=pulse` an alias for `SDL_AUDIODRIVER=pulseaudio`, avoiding the problem.

In the latest Steam update (2021-09-13, 1631572918), both beta and general-availability, the version of SDL in the `scout` runtime has received the same change.
